data-es2016plus.js

```diff
@@ -6641,6 +6641,8 @@ exports.tests = [
           var actual = [];
           var p = new Proxy({}, { get: function(o, k) { actual.push(k); return o[k]; }});
           Object.getOwnPropertyDescriptor(RegExp.prototype, 'flags').get.call(p);
+          expected.sort();
+          actual.sort();
```

Comment:
I'm confused why this fixes anything - part of the spec requirements is that the list is already alphabetized here.

Reply:
Before sorting, on Node.js 16.19.1 `expected` and `actual` look like this:
```
expected: [
  'hasIndices',
  'global',
  'ignoreCase',
  'multiline',
  'dotAll',
  'unicode',
  'sticky'
]
actual: [
  'global',
  'ignoreCase',
  'multiline',
  'dotAll',
  'unicode',
  'sticky',
  'hasIndices'
]
```
Given that the order should not matter, isn't an object a better container for these values?

Reply:
order should matter.

Reply:
The spec mandates the order.

Reply:
The specs also require that the flag `hasIndices` is implemented but not all JavaScript engines support it at this moment. That's why this compatibility table exists, isn't it?

Node.js 16 implements `hasIndices` despite not returning the flag names in the expected order.
In fact, `RegExp.prototype.flags` returns them in the expected order. It's only the property descriptor the one that lists them in a different order.

Reply:
Which means node 16 isn’t compliant.

I’d be fine splitting this test into two - one that checks the functionality and existence of the accessors, and another that checks the ordering - but spec compliance is binary, and if the order is wrong, it fails.

Reply:
I don't get it. What is so important with this order?

Reply:
Every single thing the spec mandates is equally critically important - compliance is binary, either something does, or doesn’t.